Robert/docs slashing (#8329)
cosmos/cosmos-sdk: x/slashing/spec/05_hooks.md

# Hooks

In this section we describe the "hooks" - slashing module code that runs when other events happen.

> The module provides the following hooks, which can be useful

Reviewer: Maybe I'd rephrase both lines (this and the one above) as follows:

```
This section contains a description of the module's `hooks`. Hooks are operations that are executed automatically when events are raised.
```

Author: OK, looks good

## Staking hooks

> The slashing module implements the x/staking `StakingHooks` used to keep information about validators. During the app initialization, these hooks should be registered in the staking module object.

Reviewer: ```suggestion
The slashing module implements the `StakingHooks` defined in `x/staking` and are used as record-keeping of validators information. During the app initialization, these hooks should be registered in the staking module struct.
```

The following hooks impact the slashing state:

> + `AfterValidatorBonded` - creates a `ValidatorSigningInfo` as described in the following section
> + `AfterValidatorCreated` - registers validator's consensus key
> + `AfterValidatorRemoved` - removes validator's consensus key

Reviewer: ```suggestion
+ `AfterValidatorBonded` creates a `ValidatorSigningInfo` instance as described in the following section.
+ `AfterValidatorCreated` stores a validator's consensus key.
+ `AfterValidatorRemoved` removes a validator's consensus key.
```
